### DISCUSSION TITLE
How to validate minimum length with `bubbles/textinput`?

### DISCUSSION THREAD

**Discussion by erwin** [charmbracelet/bubbles#289] [Q&A] [answered]
for fields like your `textinput` credit card example in `~/bubbletea/examples/credit-card-form/main.go` you have:

```
	inputs[cvv].Validate = cvvValidator
```

and 

```
func cvvValidator(s string) error {
	// The CVV should be a number of 3 digits
	// Since the input will already ensure that the CVV is a string of length 3,
	// All we need to do is check that it is a number
	_, err := strconv.ParseInt(s, 10, 64)
	return err
}
```

Maybe I'm just misunderstanding. It seems to me like `cvvValidator` and the forms `CharLimit = 3` will not let you enter more than 3 characters, **but I don't think it's doing anything to make sure you enter at-least 3 characters.** If you throw an error because there are only 1 or 2 characters, it's probably because the user has still not finished.

Yet if the user HAS finished and there are only 1 or 2 characters sent to `cvvValidator` then I DO want to throw an error...

![image](https://user-images.githubusercontent.com/11534217/200535097-4f75c8a2-cae3-484f-becd-3c7553bef595.png)

It's likely that there's a technique for these sorts of situations that I'm not familiar with - if you know it, your knowledge is much appreciated.

In my way of thinking, it seems like most field validation can only be completed once the user attempts to move focus to the next field.

It would be nice if the various *Style's could automatically be set to an global lipgloss.Style for errors when field validation fails...

**Comment by bashbunni** [comment, marked as answer]
Hi @erwin, 

You can definitely do validation and throw errors at any point. You just need to add a helper function to add whatever specifics you're looking for. I created a gist to show you. You can call the min length validation when the user hits enter, which is how they would indicate "done" with a field. From there you could use the result to block them from going to the next field, or just show the error message and do nothing, which is what I've done in the Gist for simplicity's sake. 

https://gist.github.com/bashbunni/fed91563900a9f6e20cde881fe68ac31

I hope that helps. Let us know if you're still unsure or if there's a part of the question I missed.

**Comment by erwin** [comment]
Thank you @bashbunni 

That is pretty much how I ended up handling it, and it works well.
